balpy/balpy.py

```diff
@@ -1510,11 +1515,195 @@ def balLinearPoolInitJoin(self, poolDescription, poolId, slippageTolerancePercen
+            bptAmount,
+            tokenOut,
+            minAmountsOut,
+            toInternalBalance=True
+        ):
+		tokenList, tokenOut, minAmountsOut = self.balSortTokensExitPoolExactBptInForOneTokenOut(
+		    tokenList, tokenOut, minAmountsOut)
+		userData = eth_abi.encode_abi(["uint256", "uint256", "uint256"], [
+		                              exitKindValue, bptAmount, tokenOut])
+		return tokenList, minAmountsOut, userData, toInternalBalance
+
+	def balSortTokensExitPoolExactBptInForTokensOut(self, tokenList, minAmountsOut):
+		tokenListAddresses = [self.web3.toChecksumAddress(
+		    token) for token in tokenList]
+		minAmountsOutSorted = [
+			minAmountOut
+			for _, minAmountOut in sorted(zip(tokenListAddresses, minAmountsOut))
+		]
+		tokenListAddressSorted = sorted(tokenListAddresses)
+		return tokenListAddressSorted, minAmountsOutSorted
+
+	def balFormatExitPoolRequestTupleExactBptInForTokensOut(
+            self,
+            exitKindValue,
+            tokenList,
+            bptAmount,
+            minAmountsOut,
+            toInternalBalance=True
+        ):
+		tokenList, minAmountsOut = self.balSortTokensExitPoolExactBptInForTokensOut(
+		    tokenList, minAmountsOut)
+		userData = eth_abi.encode_abi(["uint256", "uint256"], [
+		                              exitKindValue, bptAmount])
+		return tokenList, minAmountsOut, userData, toInternalBalance
+
+	def balSortTokensExitPoolBptInForExactTokensOut(self, tokenList, amountsOut, minAmountsOut):
+		tokenListAddresses = [self.web3.toChecksumAddress(
+		    token) for token in tokenList]
+		minAmountsOutSorted = [
+			minAmountOut
+			for _, minAmountOut in sorted(zip(tokenListAddresses, minAmountsOut))
+		]
+		amountsOutSorted = [
+			amountOut for _, amountOut in sorted(zip(tokenListAddresses, amountsOut))
+		]
+		tokenListAddressSorted = sorted(tokenListAddresses)
+		return tokenListAddressSorted, amountsOutSorted, minAmountsOutSorted
+
+	def balFormatExitPoolRequestTupleBptInForExactTokensOut(
+                self,
+                exitKindValue,
+                tokenList,
+                maxBptAmount,
+                amountsOut,
+                minAmountOut,
+                toInternalBalance=True):
+		amountsOut = self.balConvertTokensToWei(tokenList, amountsOut)
+		tokenList, amountsOut, minAmountOut = self.balSortTokensExitPoolBptInForExactTokensOut(
+		    tokenList, amountsOut, minAmountOut)
+		userData = eth_abi.encode_abi(["uint256", "uint256[]", "uint256"], [
+		                              exitKindValue, amountsOut, maxBptAmount])
+		return tokenList, minAmountOut, userData, toInternalBalance
+
+	def balExitPool(
+            self,
+            poolId: str,
+            bptAmount: float,
+            tokenList: List[str],
+            exitKind: str,
+            tokenOut: int = None,
+            amountsOut: List[int] = [],
+            query: bool = False,
+            toInternalBalance: bool = True,
+            minAmountsOut: List[float] = None,
```

Comment:
please follow convention laid out by `balJoinPool` or update the rest of the repo to follow this `**exitData` pattern